Tests/AsyncKitTests/ConnectionPoolTests.swift

```diff
@@ -8,97 +8,108 @@ import NIOPosix
-        
+
         // make two connections
         let connA = try pool.requestConnection().wait()
         XCTAssertEqual(connA.isClosed, false)
         let connB = try pool.requestConnection().wait()
         XCTAssertEqual(connB.isClosed, false)
         XCTAssertEqual(foo.connectionsCreated.load(ordering: .relaxed), 2)
-        
+
         // try to make a third, but pool only supports 2
-        var connC: FooConnection?
-        pool.requestConnection().whenSuccess { connC = $0 }
-        XCTAssertNil(connC)
+        let futureC = pool.requestConnection()
+        let connC = ManagedAtomic<FooConnection?>(nil)
+        futureC.whenSuccess { connC.store($0, ordering: .relaxed) }
+        XCTAssertNil(connC.load(ordering: .relaxed))
         XCTAssertEqual(foo.connectionsCreated.load(ordering: .relaxed), 2)
-        
+
         // release one of the connections, allowing the third to be made
         pool.releaseConnection(connB)
-        XCTAssertNotNil(connC)
-        XCTAssert(connC === connB)
+        let connCRet = try futureC.wait()
+        XCTAssertNotNil(connC.load(ordering: .relaxed))
+        XCTAssert(connC.load(ordering: .relaxed) === connB)
+        XCTAssert(connCRet === connC.load(ordering: .relaxed))
         XCTAssertEqual(foo.connectionsCreated.load(ordering: .relaxed), 2)
-        
+
         // try to make a third again, with two active
-        var connD: FooConnection?
-        pool.requestConnection().whenSuccess { connD = $0 }
-        XCTAssertNil(connD)
+        let futureD = pool.requestConnection()
+        let connD = ManagedAtomic<FooConnection?>(nil)
+        futureD.whenSuccess { connD.store($0, ordering: .relaxed) }
+        XCTAssertNil(connD.load(ordering: .relaxed))
         XCTAssertEqual(foo.connectionsCreated.load(ordering: .relaxed), 2)
-        
+
         // this time, close the connection before releasing it
-        try connC!.close().wait()
-        pool.releaseConnection(connC!)
-        XCTAssert(connD !== connB)
-        XCTAssertEqual(connD?.isClosed, false)
+        try connCRet.close().wait()
+        pool.releaseConnection(connC.load(ordering: .relaxed)!)
+        let connDRet = try futureD.wait()
+        XCTAssert(connD.load(ordering: .relaxed) !== connB)
+        XCTAssert(connDRet === connD.load(ordering: .relaxed))
+        XCTAssertEqual(connD.load(ordering: .relaxed)?.isClosed, false)
         XCTAssertEqual(foo.connectionsCreated.load(ordering: .relaxed), 3)
     }
 
-    func testConnectionPrunning() async throws {
+    func testConnectionPruning() async throws {
         let foo = FooDatabase()
         let pool = EventLoopConnectionPool(
             source: foo,
             maxConnections: 5,
-            on: MultiThreadedEventLoopGroup(numberOfThreads: 1).next(),
-            pruneInterval: 0.1,
-            maxIdleTimeBeforePrunning: 0.15
+            pruneInterval: .milliseconds(200),
+            maxIdleTimeBeforePruning: .milliseconds(300),
```

Comment:
The general slowness of GitHub CI and also TSan being enabled made this test a little bit flaky, so i doubled these two.